server/src/main/java/com/linecorp/centraldogma/server/CentralDogma.java

```diff
@@ -320,13 +322,14 @@ private void doStart() throws Exception {
         ScheduledExecutorService purgeWorker = null;
         ProjectManager pm = null;
         CommandExecutor executor = null;
-        PrometheusMeterRegistry meterRegistry = null;
+        CompositeMeterRegistry meterRegistry = null;
         Server server = null;
         SessionManager sessionManager = null;
         try {
-            meterRegistry = PrometheusMeterRegistries.newRegistry();
+            meterRegistry = new CompositeMeterRegistry();
```

Comment:
Can't we just use `Flags.meterRegistry()` instead of creating `CompositeMeterRegistry` and adding it to the another `CompositeMeterRegistry`?

Reply:
> Can't we just use Flags.meterRegistry() instead of creating CompositeMeterRegistry and adding it to the another CompositeMeterRegistry?

This was the first idea I had as well, before I started running into CI failures.
https://github.com/line/centraldogma/actions/runs/3719341734

Since integration tests continuously start up and shut down the server (and shutdown is asynchronous):
1. Tests that rely on using `meterRegistry` fail since a previous `meter` with a same name has been registered
2. An `OutOfMemoryException` is raised presumably due to `meterRegistry` not being cleaned up from previous tests.

I think what you have suggested can work if:
1) We make sure that between each test we wait until the `CentralDogma` is completely stopped synchronously
2) Between each test we clear the `Flags.meterRegistry`

I'm not sure if this is a good idea though 😅   I'm welcome to ideas.

Reply:
I should've read the description carefully. Sorry about it. 😓 
We didn't have any problem with Armeria so I thought it was also going to work.
So If we stop the server synchronously, we don't have any problem using the `Flags.meterRegitry()`. Is that correct?

Reply:
> So If we stop the server synchronously, we don't have any problem using the Flags.meterRegitry(). Is that correct?

If we stop the server synchronously and clear the `meterRegistry`between each test I think it should work fine (there are also some tests which don't clean up the `CentralDogma` server so we need to handle those as well)

> We didn't have any problem with Armeria so I thought it was also going to work.

I thought so too 😅

Reply:
Alternatively, we can also just remove tests that rely on checking metrics for correctness

e.g.
https://github.com/line/centraldogma/blob/be0426bf89418328c4ba11f2ac1940173583c32a/it/src/test/java/com/linecorp/centraldogma/it/PrematureClientFactoryCloseTest.java#L81-L84

Reply:
If so, I'm leaning toward modifying the tests because:
- It doesn't make sense to implement production code inefficiently because test code doesn't work
- It probably wouldn't take so long to complete tests even if we change to close synchronously compared to Armeria build time. 😆

Reply:
I've made changes so that flakiness doesn't affect production code.

The main points were:
1) Add a setter method `CentralDogmaBuilder#meterRegistry(MeterRegistry)` so that users can specify the `MeterRegistry` to be used.
2) For the main implementation, use `Flags#meterRegistry()`.
3) For test implementations, initialize a new `CompositeMeterRegistry`

Please take another look 🙏